src/test/TestPrinter.cpp

```diff
@@ -6,14 +6,6 @@
 
 namespace Catch
 {
-
-template <>
```

Comment:
this is useful for checking why tests failed in test outputs - it is not used directly, so removing it does not fail compilation

why is this removed?

Reply:
I had to remove it because Visual C++ fails with this code at link time because it says it's a duplicate symbol. I will create a separate issue to look into this with the error output.

Reply:
nm, found the problem: standard says that template specializations have to be defined before the first instantiation of the template (or else end up with multiple implementations specific to the instantiation sites). By including TestPrinter.h in the right places, it resolved this issue